### PR TITLE
support for GlobalRole

### LINKED ISSUE #370: Ability to define global scope roles in rancher2_role_template
As custom global role creation was introduced in Rancher v2.4.0, support for this should be added in the rancher2 rancher2_role_template resource https://www.terraform.io/docs/providers/rancher2/r/roleTemplate.html

gz#10716

gz#13391

### PR DESCRIPTION
**Pull request by jasonhancock**
fixes rancher/terraform-provider-rancher2#370

### CODE REVIEW

File: docs/data-sources/global_role.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+* `name` - (Required) The name of the Global Role (string)
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - (Computed) The ID of the resource (string)
+* `builtin` - (Computed) Builtin global role (bool)
+* `description` - (Computed) Global role description (string)
+* `new_user_default` - (Computed) Whether or not this role should be added to new users (bool)
```
Comment (rawmind0) [inline docs/data-sources/global_role.md:30]: `new_user_default` argument is not defined as computed on dataSource schema definition `globalRoleFields()`

File: rancher2/data_source_rancher2_global_role.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceRancher2GlobalRole() *schema.Resource {
+	return &schema.Resource{
+		Read:   dataSourceRancher2GlobalRoleRead,
+		Schema: globalRoleFields(),
```
Comment (rawmind0) [inline rancher2/data_source_rancher2_global_role.go]: DataSource and Resource shouldn't share same schema as arguments properties should be different. All dataSource arguments but `name` should be computed.

File: rancher2/schema_global_role.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package rancher2
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+//Schemas
+
+func globalRoleFields() map[string]*schema.Schema {
```
Comment (rawmind0) [inline rancher2/schema_global_role.go:9]: As mentioned before, `globalRoleFields()` should be different from dataSource and Resource. Could you please add another function to define dataSource flelds (here or in `rancher2/data_source_rancher2_global_role.go` file)??

Something like,
```
func dataSourceGlobalRoleFields() map[string]*schema.Schema {
	s := map[string]*schema.Schema{
		"builtin": {
			Type:        schema.TypeBool,
			Computed:    true,
			Description: "Builtin global role",
		},
		"description": {
			Type:        schema.TypeString,
			Computed:    true,
			Description: "Global role policy description",
		},
		"name": {
			Type:        schema.TypeString,
			Required:    true,
			Description: "Global role policy name",
		},
		"new_user_default": {
			Type:        schema.TypeBool,
			Computed:    true,
			Description: "Whether or not this role should be added to new users",
		},
		"rules": {
			Type:        schema.TypeList,
			Computed:    true,
			Description: "Global role policy rules",
			Elem: &schema.Resource{
				Schema: policyRuleFields(),
			},
		},
                "annotations": {
	  	        Type:     schema.TypeMap,
			Computed: true,
		},
		"labels": {
			Type:     schema.TypeMap,
			Computed: true,
		},
	}

	return s
}
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]
Hi @jasonhancock , thanks for this PR.

Could you please take a look to the requested changes?? Additionally, could you also please update CHANGELOG.md accordingly??

**Comment by jasonhancock** [comment]
@rawmind0 can you let me know if I missed anything? Thanks!

**Review by rawmind0 — APPROVED** [review]

**Comment by rawmind0** [comment]
Thanks @jasonhancock 

LGTM

**Pull request merged**